File: ruby/spec/problem002_spec.rb
require 'problem002'

# https://projecteuler.net/index.php?section=problems&id=2

describe "Euler Problem #2" do
  it "Solves #1" do
    expect(euler_2(1)).to eq 0
    expect(euler_2(2)).to eq 2
    expect(euler_2(3)).to eq 2
    expect(euler_2(4)).to eq 2
    expect(euler_2(5)).to eq 2
    expect(euler_2(6)).to eq 2
    expect(euler_2(7)).to eq 2
    expect(euler_2(8)).to eq 10
    expect(euler_2(9)).to eq 10
    expect(euler_2(10)).to eq 10
    expect(euler_2(4000000)).to eq 4613732
  end
end

File: clojure/spec/euler/level1/problem002_spec.clj
(ns euler.level1.problem002-spec
  (:require
    [speclj.core :refer :all]
    [euler.level1.problem002 :refer :all]))

; https://projecteuler.net/problem=2

(describe "Euler Problem #2"

  (it "even-fibs"
    (should= [2] (take 1 (even-fibs)))
    (should= [2 8] (take 2 (even-fibs)))
    (should= [2 8 34] (take 3 (even-fibs))))

  (it "even-fibs-less-than"
    (should= [] (even-fibs-less-than 0))
    (should= [] (even-fibs-less-than 1))
    (should= [] (even-fibs-less-than 2))
    (should= [2] (even-fibs-less-than 3))
    (should= [2] (even-fibs-less-than 4))
    (should= [2] (even-fibs-less-than 5))
    (should= [2] (even-fibs-less-than 6))
    (should= [2] (even-fibs-less-than 7))
    (should= [2] (even-fibs-less-than 8))
    (should= [2 8] (even-fibs-less-than 9))
    (should= [2 8 34] (even-fibs-less-than 100)))

  (it "Solves #2"
    (should= 2 (euler-2 2))
    (should= 10 (euler-2 8))
    (should= 4613732 (euler-2 4000000))))

